### PR TITLE
Test 3941

### PR DESCRIPTION
**Pull request by danylohlynsky**
Added tests for methods:
ChatMessageServiceImpl.deleteMessage
ChatMessageServiceImpl.likeMessage
ChatRoomServiceImpl.mapListChatMessageDto
ChatRoomServiceImpl.findPrivateByParticipantsForSocket

### CODE REVIEW

**Comment by danylohlynsky** [comment]
Added tests for:
ChatRoomServiceImpl.findAllByParticipantName
ChatRoomServiceImpl.createNewChatRoom
Tests that was in cart but was already done checked:
 ChatRoomDtoMapper.convert
ChatMessageMapper.convert

**Review by Pavl0-Gural — APPROVED** [review]

**Review by Andrii-Yezenitskyi — APPROVED** [review]

**Review by Max877866 — APPROVED** [review]

File: chat/pom.xml [hunk 1]
```diff
@@ -149,6 +149,12 @@
             <artifactId>azure-storage-blob</artifactId>
             <version>12.7.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
+            <artifactId>powermock-api-mockito2</artifactId>
+            <version>2.0.7</version>
```
Comment (thesvider) [inline chat/pom.xml:155]: in general, it is good practice to have properties for dependency versions.

File: chat/src/test/java/greencity/service/impl/ChatMessageServiceImplTest.java [hunk 2]
```diff
@@ -147,4 +145,58 @@ void processMessage() {
...
+    }
+
+    @Test
+    void deleteLikeFromMessage() {
+        MessageLike messageLike = new MessageLike(1L, 1L);
+        when(chatMessageRepo.getParticipantIdIfLiked(1L, 1L)).thenReturn(1L);
+        ChatMessage chatMessage = ChatMessage.builder()
+            .id(1L)
+            .content("Content")
+            .createDate(ZonedDateTime.of(2022, 12, 12, 12, 12, 12, 12, ZoneId.systemDefault()))
```
Comment (thesvider) [inline chat/src/test/java/greencity/service/impl/ChatMessageServiceImplTest.java:167]: why just not to have ZonedDateTime.now() ?

File: chat/src/test/java/greencity/service/impl/ChatRoomServiceImplTest.java [hunk 3]
```diff
@@ -99,17 +111,23 @@ void init() {
             .chatType(ChatType.PRIVATE)
             .participants(new HashSet<>())
             .build();
+        expectedChatMessageList.add(ChatMessage.builder().id(1L).room(expected).sender(expectedParticipant).build());
+        expectedListDto.add(ChatRoomDto.builder().id(1L).build());
```
Comment (thesvider) [inline chat/src/test/java/greencity/service/impl/ChatRoomServiceImplTest.java:115]: create private method that accepts argument as id and have List.of(buildChatRoom(1L), buildChatRoom(2L) )

File: chat/src/test/java/greencity/service/impl/ChatRoomServiceImplTest.java [hunk 4]
```diff
@@ -262,11 +290,58 @@ public void findAllChatRoomsByQuery() {
...
+    void addNewUserToSystemChat() {
         Long id = 1L;
         when(chatRoomRepo.findSystemChatRooms()).thenReturn(Collections.singletonList(expected));
         chatRoomService.addNewUserToSystemChat(id);
         verify(chatRoomRepo).addUserToSystemChatRoom(expected.getId(), id);
     }
 
+    @Test
+    void mapListChatMessageDto()
+        throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
```
Comment (thesvider) [inline chat/src/test/java/greencity/service/impl/ChatRoomServiceImplTest.java:302]: no need to have these exceptions. use @ SneakyThrpws annotation instead.

**Pull request merged**